ts/screens/main/MessagesScreen.tsx

```diff
@@ -1,24 +1,87 @@
-import { Container } from "native-base";
+import { Container, Content, Text } from "native-base";
 import * as React from "react";
-import { NavigationScreenProp, NavigationState } from "react-navigation";
+import { ActivityIndicator } from "react-native";
+import {
+  NavigationEventSubscription,
+  NavigationScreenProp,
+  NavigationState
+} from "react-navigation";
+import { connect } from "react-redux";
+
+import { MessageWithContent } from "../../../definitions/backend/MessageWithContent";
+import { ReduxProps } from "../../actions/types";
+import { GlobalState } from "../../reducers/types";
+import { ServicesListObject } from "../../sagas/messages";
+import { loadMessages } from "../../store/actions/messages";
+import { orderedMessagesSelector } from "../../store/reducers/entities/messages";
+import { createLoadingSelector } from "../../store/reducers/loading";
+
+type ReduxMappedProps = {
+  isLoadingMessages: boolean;
+  messages: ReadonlyArray<MessageWithContent>;
+  servicesById: ServicesListObject;
+};
 
 export type OwnProps = {
   navigation: NavigationScreenProp<NavigationState>;
 };
 
-export type Props = OwnProps;
+export type Props = ReduxMappedProps & ReduxProps & OwnProps;
 
 /**
  * This screen show the messages to the authenticated user.
  */
 class MessagesScreen extends React.Component<Props, never> {
+  private didFocusSubscription:
+    | NavigationEventSubscription
+    | undefined = undefined;
+
   constructor(props: Props) {
     super(props);
   }
 
+  public componentDidMount() {
+    // tslint:disable-next-line
+    this.didFocusSubscription = this.props.navigation.addListener(
+      "didFocus",
+      () => {
+        this.loadMessages();
+      }
+    );
+  }
+
+  public componentWillUnmount() {
+    if (this.didFocusSubscription) {
+      this.didFocusSubscription.remove();
+    }
+  }
+
+  private renderLoadingStatus = (isLoadingMessages: boolean) => {
+    return isLoadingMessages ? (
+      <ActivityIndicator size="small" color="#00ff00" />
```

Comment:
it would be nice to provide some kind of progress (e.g. loaded X of Y messages) - pls add a story for implementing this later